exporters/trace/src/main/java/com/google/cloud/opentelemetry/trace/TraceExporter.java

```diff
@@ -15,134 +15,115 @@
-import com.google.api.gax.grpc.GrpcTransportChannel;
-import com.google.api.gax.rpc.FixedTransportChannelProvider;
-import com.google.api.gax.rpc.HeaderProvider;
-import com.google.auth.Credentials;
-import com.google.auth.oauth2.GoogleCredentials;
-import com.google.cloud.trace.v2.TraceServiceClient;
-import com.google.cloud.trace.v2.TraceServiceSettings;
-import com.google.cloud.trace.v2.stub.TraceServiceStub;
-import com.google.common.collect.ImmutableMap;
-import com.google.devtools.cloudtrace.v2.AttributeValue;
-import com.google.devtools.cloudtrace.v2.ProjectName;
-import com.google.devtools.cloudtrace.v2.Span;
-import io.grpc.ManagedChannelBuilder;
+import com.google.cloud.ServiceOptions;
 import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.internal.ThrottlingLogger;
 import io.opentelemetry.sdk.trace.data.SpanData;
 import io.opentelemetry.sdk.trace.export.SpanExporter;
 import java.io.IOException;
-import java.util.ArrayList;
 import java.util.Collection;
-import java.util.List;
-import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.Nonnull;
 
 public class TraceExporter implements SpanExporter {
 
-  private final CloudTraceClient cloudTraceClient;
-  private final ProjectName projectName;
-  private final String projectId;
-  private final TraceTranslator translator;
+  private static final Logger logger = Logger.getLogger(TraceExporter.class.getName());
 
-  private static final Map<String, String> HEADERS =
-      Map.of("User-Agent", "opentelemetry-operations-java/" + TraceVersions.EXPORTER_VERSION);
-  private static final HeaderProvider HEADER_PROVIDER = () -> HEADERS;
+  private final TraceConfiguration.Builder customTraceConfigurationBuilder;
+  private final AtomicReference<SpanExporter> internalTraceExporter;
+  private final ThrottlingLogger throttlingLogger;
 
-  public static SpanExporter createWithDefaultConfiguration() throws IOException {
-    TraceConfiguration configuration = TraceConfiguration.builder().build();
-    return TraceExporter.createWithConfiguration(configuration);
+  private TraceExporter(TraceConfiguration.Builder configurationBuilder) {
+    this.customTraceConfigurationBuilder = configurationBuilder;
+    this.internalTraceExporter = new AtomicReference<>(null);
+    this.throttlingLogger = new ThrottlingLogger(logger);
   }
 
-  public static SpanExporter createWithConfiguration(TraceConfiguration configuration)
-      throws IOException {
-    String projectId = configuration.getProjectId();
-    TraceServiceStub stub = configuration.getTraceServiceStub();
-
-    // TODO: Remove stub.
-    if (stub == null) {
-      TraceServiceSettings.Builder builder = TraceServiceSettings.newBuilder();
+  private static SpanExporter generateStubTraceExporter(TraceConfiguration.Builder configBuilder) {
```

Comment:
Are you using "stub" to mean something like "placeholder" or "temporary" here? If so, consider a different name because "stub" is generally used to mean [RPC stub](https://en.wikipedia.org/wiki/Stub_(distributed_computing)).

Reply:
Yes, I was indeed using it for that. I can change the naming then.

Reply:
For lack of a better name, replaced it with `generateSkeletonTraceExporter`, other alternative considered was `generateTraceExporterTemplate`, but it seemed to me it was not clarifying the semantics well enough. But I am open to suggestions on this.